IBM Cloud Shell data resource

<!---
Please note the following potential times when an issue might be in Terraform core:

* [Configuration Language](https://www.terraform.io/docs/configuration/index.html) or resource ordering issues
* [State](https://www.terraform.io/docs/state/index.html) and [State Backend](https://www.terraform.io/docs/backends/index.html) issues
* [Provisioner](https://www.terraform.io/docs/provisioners/index.html) issues
* [Registry](https://registry.terraform.io/) issues
* Spans resources across multiple providers

If you are running into one of these scenarios, we recommend opening an issue in the [Terraform core repository](https://github.com/hashicorp/terraform/) instead.
--->

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

### Terraform CLI and Terraform IBM Provider Version

<!--- Please run `terraform -v` to show the Terraform core version and provider version(s). If you are not running the latest version of Terraform or the provider, please upgrade because your issue may have already been fixed. [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions). --->

```
Terraform v1.0.10
on darwin_amd64
+ provider registry.terraform.io/eiqops/restapi v1.15.1
+ provider registry.terraform.io/ibm-cloud/ibm v1.34.0
```
### Affected Resource(s)

<!--- Please list the affected resources and data sources. --->

* ibm_cloud_shell_account_settings

### Terraform Configuration Files



```hcl
terraform {
  required_version = ">= 1.0.7"
  required_providers {
    ibm = {
      source  = "IBM-Cloud/ibm"
      version = ">= 1.34.0"
      # version = "= 1.26.2"

    }
  }
}

provider "ibm" {
  ibmcloud_api_key = var.ibmcloud_api_key
  region           = "us-south"
}

// Data source to get account settings
data "ibm_iam_account_settings" "iam_account_settings" {
}

data "ibm_cloud_shell_account_settings" "cloud_shell_account_settings" {
  account_id = data.ibm_iam_account_settings.iam_account_settings.account_id
}

output "account_id" {
  value = data.ibm_iam_account_settings.iam_account_settings.account_id
}

```

### Debug Output

<!---
Please provide a link to a GitHub Gist containing the complete debug output. Please do NOT paste the debug output in the issue; just paste a link to the Gist.

To obtain the debug output, see the [Terraform documentation on debugging](https://www.terraform.io/docs/internals/debugging.html).
--->
https://gist.github.com/rjminsha/66364a4d7b14821b04cb049a37b6ca60

### Panic Output

<!--- If Terraform produced a panic, please provide a link to a GitHub Gist containing the output of the `crash.log`. --->

### Expected Behavior

<!--- What should have happened? --->
expected to get the shell resource

### Actual Behavior

<!--- What actually happened? --->
```
2021-10-29T11:37:58.248+1300 [ERROR] plugin.(*GRPCProvider).ReadDataSource: error="rpc error: code = Unavailable desc = transport is closing"
╷
│ Error: Plugin did not respond
│ 
│   with data.ibm_cloud_shell_account_settings.cloud_shell_account_settings,
│   on main.tf line 23, in data "ibm_cloud_shell_account_settings" "cloud_shell_account_settings":
│   23: data "ibm_cloud_shell_account_settings" "cloud_shell_account_settings" {
│ 
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ReadDataSource call. The plugin logs may contain more
│ details.
╵

Stack trace from the terraform-provider-ibm_v1.34.0 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x2862217]

goroutine 82 [running]:
github.com/IBM-Cloud/terraform-provider-ibm/ibm.dataSourceIBMCloudShellAccountSettingsRead(0x3bf8118, 0xc0013c08a0, 0xc000056e80, 0x36b3a00, 0xc000612800, 0xc0006a2dd0, 0xc000791948, 0x100dfd8)
``` 
### Steps to Reproduce

<!--- Please list the steps required to reproduce the issue. --->

1. `terraform apply`

### Important Factoids

<!--- Are there anything atypical about your accounts that we should know? For example: Running in EC2 Classic? --->

### References

<!---
Information about referencing Github Issues: https://help.github.com/articles/basic-writing-and-formatting-syntax/#referencing-issues-and-pull-requests

Are there any other GitHub issues (open or closed) or pull requests that should be linked here? Vendor documentation? For example:
--->

* #0000

--- a/ibm/data_source_ibm_cloud_shell_account_settings.go
+++ b/ibm/data_source_ibm_cloud_shell_account_settings.go
@@ -122,12 +122,12 @@ func dataSourceIBMCloudShellAccountSettingsRead(context context.Context, d *sche
 	getAccountSettingsOptions.SetAccountID(d.Get("account_id").(string))
 
 	accountSettings, response, err := ibmCloudShellClient.GetAccountSettingsWithContext(context, getAccountSettingsOptions)
-	if err != nil {
+	if err != nil || accountSettings == nil {
 		log.Printf("[DEBUG] GetAccountSettingsWithContext failed %s\n%s", err, response)
 		return diag.FromErr(fmt.Errorf("GetAccountSettingsWithContext failed %s\n%s", err, response))
 	}
 
-	d.SetId(*accountSettings.ID)
+	d.SetId(*accountSettings.AccountID)
 	if err = d.Set("rev", accountSettings.Rev); err != nil {
 		return diag.FromErr(fmt.Errorf("Error setting rev: %s", err))
 	}

